src/autoschedulers/adams2019/cost_model_generator.cpp

```diff
@@ -71,7 +80,7 @@ struct ModelWeight<true> : public GeneratorInput<Buffer<float>> {
         Expr smoothed_second_moment_correction = 1 / (1 - pow(0.999f, timestep + 1));
 
         // Update the weights
-        Expr step = learning_rate * smoothed_deriv * smoothed_deriv_correction;
+        Expr step = std::move(learning_rate) * smoothed_deriv * smoothed_deriv_correction;
```

Comment:
I think the changes in this file (particularly the std::moves) are harmful for readability. Several of our clang checks are (very very *very* slightly) good for performance but bad for readability, and they shouldn't be on in non-performance-critical code (which includes just about all of use-of-Halide-as-a-language code)

Reply:
...wtf? yikes. I didn't even notice that (I never thought to check for that sort of craziness). Yeah, that's some weird stuff there. Let me see what I can do.